bmm/src/main/java/org/openehr/bmm/v2/persistence/PBmmGenericProperty.java

```diff
@@ -1,5 +1,29 @@
 package org.openehr.bmm.v2.persistence;
 
+import org.openehr.bmm.core.BmmClass;
+import org.openehr.bmm.core.BmmGenericType;
+import org.openehr.bmm.core.BmmUnitaryProperty;
+import org.openehr.bmm.v2.validation.converters.BmmClassProcessor;
+
 public final class PBmmGenericProperty extends PBmmProperty<PBmmGenericType> {
 
+    public PBmmGenericProperty() {
+        super();
+    }
+
+    @Override
+    public BmmUnitaryProperty createBmmProperty(BmmClassProcessor classProcessor, BmmClass bmmClass) {
+        PBmmGenericType typeRef = getTypeRef();
+        if (typeRef != null) {
+            BmmGenericType bmmType = typeRef.createBmmType(classProcessor, bmmClass);
+            if (bmmType != null) {
+                BmmUnitaryProperty bmmProperty = new BmmUnitaryProperty(getName(), bmmType, getDocumentation(), nullToFalse(isMandatory()), nullToFalse(isComputed()));
+                populateImBooleans(bmmProperty);
+                return bmmProperty;
+            }
```

Comment:
`createBmmType` always returns a non-null value, so this if-statement can be removed.

```suggestion
            BmmGenericType bmmType = typeRef.createBmmType(classProcessor, bmmClass);
            BmmUnitaryProperty bmmProperty = new BmmUnitaryProperty(getName(), bmmType, getDocumentation(), nullToFalse(isMandatory()), nullToFalse(isComputed()));
            populateImBooleans(bmmProperty);
            return bmmProperty;
```